google/cloud/aiplatform/_pipeline_based_service/pipeline_based_service.py

```diff
@@ -0,0 +1,328 @@
+
+        if not display_name:
+            display_name = cls._generate_display_name()
+
+        self = cls._empty_constructor(
+            project=project,
+            location=location,
+            credentials=credentials,
+        )
+
+        service_pipeline_job = pipeline_jobs.PipelineJob(
+            display_name=display_name,
+            template_path=template_path,
+            job_id=job_id,
+            pipeline_root=pipeline_root,
+            parameter_values=template_params,
+            encryption_spec_key_name=encryption_spec_key_name,
+            project=project,
+            location=location,
+            credentials=credentials,
+        )
+
+        # Suppresses logs from PipelineJob
+        # The class implementing _VertexAiPipelineBasedService should define a custom log message
+        logging.getLogger("google.cloud.aiplatform.pipeline_jobs").setLevel(
+            logging.WARNING
+        )
+
+        service_pipeline_job.submit(
+            service_account=service_account,
+            network=network,
+            experiment=experiment,
+        )
+
+        logging.getLogger("google.cloud.aiplatform.pipeline_jobs").setLevel(
+            logging.INFO
+        )
+
+        self._gca_resource = service_pipeline_job.gca_resource
+
+        return self
+
+    @classmethod
+    def list(
+        cls,
+        project: Optional[str] = None,
+        location: Optional[str] = None,
+        credentials: Optional[str] = None,
+    ) -> List["pipeline_jobs.PipelineJob"]:
+        """Lists all PipelineJob resources associated with this Pipeline Based service.
+        Args:
+            project (str):
+                Optional. The project to retrieve the Pipeline Based Services from. If not set,
+                the project set in aiplatform.init will be used.
+            location (str):
+                Optional. Location to retrieve the Pipeline Based Services from. If not set,
+                location set in aiplatform.init will be used.
+            credentials (auth_credentials.Credentials):
+                Optional. Custom credentials to use to retrieve the Pipeline Based Services from.
+                Overrides credentials set in aiplatform.init.
+        Returns:
+            (List[PipelineJob]):
+                A list of PipelineJob resource objects.
+        """
+        self = cls._empty_constructor(
+            project=project,
+            location=location,
+            credentials=credentials,
+        )
+
+        all_pipeline_jobs = pipeline_jobs.PipelineJob.list(
+            project=project,
+            location=location,
+            credentials=credentials,
+        )
+
+        service_pipeline_jobs = []
+
+        for job in all_pipeline_jobs:
+            try:
+                self._validate_pipeline_template_matches_service(job)
+                service_pipeline_jobs.append(job)
+
+            except ValueError:
+                continue
+
+        return service_pipeline_jobs
```

Comment:
I notice that `wait` is defined in the model eval subclass. Should that implementation be moved here?

Reply:
Good catch, moved it to this class.